packages/serve/src/parseArgs.ts

```diff
@@ -38,6 +38,13 @@ export default function parseArgs(cli: WebpackCLIType, args: string[]): ArgsType
     const parsedWebpackArgs = cli.argParser(core, parsedDevServerArgs.unknownArgs, true, process.title);
     const webpackArgs = parsedWebpackArgs.opts;
 
+    // Add WEBPACK_SERVE environment variable
+    if (webpackArgs.env) {
+        webpackArgs.env.WEBPACK_SERVE = true;
```

Comment:
We should check what it is `serve` command

Reply:
I didn't get what you said. Can you explain a bit more?

Reply:
We should add `WEBPACK_SERVE` to env only when we run `webpack serve` command

Reply:
`serve` package would only be accessed by users when we run `serve` command. As this code is in serve package, I don't think we need to check again. Or I am missing something ?

Reply:
nope, this will only run during serve command invocation, we're good here

Reply:
oh, yes, my mistake